common/thorhelper/thorrparse.cpp

```diff
@@ -3063,12 +3073,14 @@ RegexPattern * deserializeRegex(MemoryBuffer & in)
         case ThorRegexAsciiI:       next = new RegexAsciiIPattern();                    break;
         case ThorRegexAsciiSet:     next = new RegexAsciiSetPattern();                  break;
         case ThorRegexAsciiISet:    next = new RegexAsciiISetPattern();                 break;
+#ifdef _USE_ICU
         case ThorRegexUnicode:      next = new RegexUnicodePattern();                   break;
         case ThorRegexUnicodeI:     next = new RegexUnicodeIPattern();                  break;
         case ThorRegexUnicodeSet:   next = new RegexUnicodeSetPattern();                break;
         case ThorRegexUtf8:         next = new RegexUtf8Pattern();                      break;
         case ThorRegexUtf8I:        next = new RegexUtf8IPattern();                     break;
 //      case ThorRegexUnicodeISet:  next = new RegexUnicodeISetPattern();               break;
+#endif
```

Comment:
default should be changed to

    throw MakeStringException(99, "System was built without Unicode support");